Content.Server/Explosion/TriggerSystem.cs

```diff
@@ -72,12 +84,16 @@ public void Explode(EntityUid uid, ExplosiveComponent component)
 
         #region Flash
         private void HandleFlashTrigger(EntityUid uid, FlashOnTriggerComponent component, TriggerEvent args)
-                {
-                    if (component.Flashed) return;
+        {
+            bool repeatable = component.Repeating && component.LastFlash + TimeSpan.FromSeconds(component.Cooldown) < _gameTiming.CurTime;
 
-                    FlashableComponent.FlashAreaHelper(component.Owner, component.Range, component.Duration);
-                    component.Flashed = true;
-                }
+            if (!component.Flashed ^ repeatable)
+            {
+                FlashableComponent.FlashAreaHelper(component.Owner, component.Range, component.Duration);
+                component.Flashed = true;
+                component.LastFlash = _gameTiming.CurTime;
+            }
+        }
```

Comment:
You should set a timer once something's colliding (or alternatively have triggersystem track its own list) and then re-check if there's still a contact between the bodies and if so trigger again.